internal/worker/rundynamic.go

```diff
@@ -50,9 +54,20 @@ func RunDynamicAnalysis(pkg *pkgecosystem.Pkg, sbOpts []sandbox.Option) (result.
 	var lastStatus analysis.Status
 	var lastError error
 	for _, phase := range pkg.Manager().RunPhases() {
-		result, err := dynamicanalysis.Run(sb, pkg.Command(phase))
+		startTime := time.Now()
+		phaseResult, err := dynamicanalysis.Run(sb, pkg.Command(phase))
 		lastRunPhase = phase
 
+		runDuration := time.Since(startTime)
+		log.Info("Dynamic analysis phase finished",
+			log.Label("ecosystem", pkg.EcosystemName()),
+			log.Label("name", pkg.Name()),
+			log.Label("version", pkg.Version()),
+			log.Label("phase", string(phase)),
+			log.Label("error", strconv.FormatBool(err != nil)),
+			log.Label("dynamic_analysis_phase_duration_sec", fmt.Sprintf("%.1f", runDuration.Seconds())),
```

Comment:
This probably should be a regular key, value field, not wrapped in `log.Label`.

i.e. `"phase_duration", runDuration`

The docs aren't totally clear, but `log.Label` is used to add labels to the GCP's logging. There are cardinality limits as per https://cloud.google.com/logging/docs/logs-based-metrics/labels.

In fact, thinking about this now, our use of log.Label hasn't been particularly careful in the past :grimacing:

Reply:
I've removed all the log.Label usages